client/settings/payment-settings/account-keys-connection-status.js

```diff
@@ -0,0 +1,180 @@
+`;
+
+export const AccountKeysConnectionStatus = ( { formRef } ) => {
+	const {
+		isTesting,
+		isValid,
+		updateIsTestingAccountKeys,
+		updateIsValidAccountKeys,
+	} = useAccountKeys();
+
+	const handleTestConnection = async ( ref ) => {
+		updateIsTestingAccountKeys( true );
+		updateIsValidAccountKeys( null );
+
+		// Grab the HTMLCollection of elements of the HTML form, convert to array.
+		const elements = Array.from( ref.current.elements );
+		// Convert HTML elements array to an object acceptable for saving keys.
+		const keysToSave = elements.reduce( ( dict, ele ) => {
+			return Object.assign( dict, { [ ele.name ]: ele.value } );
+		}, {} );
+
+		let publishableKey;
+		let secretKey;
+
+		const isTestingLiveConnection =
+			keysToSave.publishable_key && keysToSave.secret_key;
+		const isTestingTestConnection =
+			keysToSave.test_publishable_key && keysToSave.test_secret_key;
+		if ( isTestingLiveConnection ) {
+			publishableKey = keysToSave.publishable_key;
+			secretKey = keysToSave.secret_key;
+		} else if ( isTestingTestConnection ) {
+			publishableKey = keysToSave.test_publishable_key;
+			secretKey = keysToSave.test_secret_key;
+		} else {
+			updateIsTestingAccountKeys( false );
+			updateIsValidAccountKeys( false );
+			return;
+		}
+
+		try {
+			const stripe = await loadStripe( publishableKey );
```

Comment:
Good call. But as mentioned in a [previous comment](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2273#discussion_r783249873), we could display a notification when an account key is entered in the wrong tab (live/test):

![image](https://user-images.githubusercontent.com/6844516/149178188-df1bfecd-8d7e-4e62-85ad-7e14a6f738c4.png)

![image](https://user-images.githubusercontent.com/6844516/149178313-8396a1b5-29d2-4912-93de-45d820856f02.png)

What are your thoughts on this approach?

Reply:
I agree the inline space is not enough, and this notification looks good to me.

Is the notification icon something we already use, or are you adding it now? I'm asking because it's different from the error icon on the connection status component.

Reply:
It is not something we already use and I am adding it now. It's an emoji:  ❌

I haven't seen the notification `icon` property used elsewhere, other than in [this notification](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/data/account/actions.js#L68).

Should I just remove the icon, in this case?

Reply:
My only concern with the emoji is that it could be inconsistent with other notifications and that it looks different in other operating systems https://emojipedia.org/cross-mark/.

> Should I just remove the icon, in this case?

If we have other error messages without any icon, I'd vote for removing it, but if not, then we would possibly want a visual indication that this is an error. Then I'd ask Jarek or David about it.